src/interfaces/registry.ts

```diff
@@ -4,29 +4,44 @@
 
 import { RegistryListing } from 'computable/dist/interfaces'
 import { Nos } from 'computable/dist/types'
+import { Map } from '../interfaces'
+
+interface Registry {
+  // deployed address of the registry contract, can be falsy at first
+  address?: string
+  applicants?: Applicant[]
+  challenges?: Challenge[]
+  listings?: Listing[]
+}
 
 // our shape for applicant is simply the 3 args required for the apply method
-export interface Applicant {
-  name: string; // this can be just about anynthing. NOTE: will need to be hashed on calling registry.apply by using the helper `stringToBytes`
-  deposit?: Nos; // will default to the paramaterizer MIN_DEPOSIT if excluded
-  data?: string; // again, just about anything can go here.
+interface Applicant {
+  name: string // this can be just about anynthing. NOTE: will need to be hashed on calling registry.apply by using the helper `stringToBytes`
+  deposit?: Nos // will default to the paramaterizer MIN_DEPOSIT if excluded
+  data?: string // again, just about anything can go here.
 }
 
-export interface Challenge {
+interface ApplicantData {
+  source?: string
+  value: Map|string
+}
+
+interface Challenge {
 
 }
 
 // the registry listing in computable does not keep the generated hash
 // of the "name". We want to have it here however to reference with that
 // name that exists on any Applications
-export interface Listing extends RegistryListing {
-  hash: string;
+interface Listing extends RegistryListing {
+  hash: string
 }
 
-export interface Registry {
-  // deployed address of the registry contract, can be falsy at first
-  address?: string;
-  applicants?: Applicant[];
-  challenges?: Challenge[];
-  listings?: Listing[];
+export {
+  Registry,
+  Applicant,
+  ApplicantData,
```

Comment:
ah, nvm

Reply:
Yeah, I tend to batch my exports at the bottom of the file. I think it makes it easier to see which parts of a module are being exported at a glance.